src/components/Nodes/wrapNode.tsx

```diff
@@ -121,15 +130,17 @@ export default (NodeComponent: ComponentType<NodeProps>) => {
       [id, selected, selectNodesOnDrag, isSelectable, onNodeDragStart]
     );
 
+    // As one of the props passed to a custom node
+    const [dragging, setDragging] = useState(false);
+
     const onDrag = useCallback(
       (event: UseDragEvent, dragPos: UseDragData) => {
-        updateNodePosition({ id, dragging: true, diff: { x: dragPos.dx, y: dragPos.dy } });
-
+        updateNodePosition({ id, diff: { x: dragPos.dx, y: dragPos.dy } });
+        setDragging(true);
```

Comment:
could you set this in the `onDragStart` handler?

Reply:
I think we can set it in `onDragStart` to be consistent with the concept of "starting the drag action".
I just think doing this in `onDrag` has the benefit of letting users play with this state in a custom node implementation more easily. 

To elaborate with an example:
When a user clicks on a node with both `onDragStart` and `onNodeClick` defined, these are the possibilities with the `d3-drag` implementation :
If the user moves the mouse upon clicking:
`mousedown` fires, `mousemove` fires and then `mouseup` fires. We call `onDragStart`, `onDrag` and `onDragEnd`.
If the user doesnt move the mouse upon clicking:
`mousedown` fires, `mouseup` fires and then `click` fires. We call `onDragStart` and `onClick`. 

So there is this intersection zone caused by `onmousedown` firings for `onDragStart` and `onClick`. But its kinda expected IMO (or we can try to find a way to diff them in the future ? ). So if we want to be consistent with the events, we can set the state in `onDragStart` but I stated my concern about the usage of it above.